src/napari/_qt/_qapp_model/qactions/_layerlist_context.py

```diff
@@ -118,29 +118,33 @@ def _paste_spatial_from_clipboard(ll: LayerList) -> None:
         return
 
     for layer in ll.selection:
-        for key in loaded:
-            loaded_attr_value = loaded[key]
-            if key == 'units':
-                loaded_attr_value = loaded_attr_value[-layer.ndim :]
-            elif isinstance(loaded_attr_value, list):
-                loaded_attr_value = np.array(loaded_attr_value)
-            if key == 'shear':
-                loaded_attr_value = loaded_attr_value[
-                    -(layer.ndim * (layer.ndim - 1)) // 2 :
-                ]
-            elif key == 'affine':
-                loaded_attr_value = loaded_attr_value[
-                    -(layer.ndim + 1) :, -(layer.ndim + 1) :
-                ]
-            elif isinstance(loaded_attr_value, np.ndarray):
-                if loaded_attr_value.ndim == 1:
+        with layer._block_refresh():
```

Comment:
@brisvag You may see here where I want to use block of refresh until all properties are updated as calling refresh triggers multiple heavy computation. 

I intend to avoid refresh, so force is exactly this that I do not want to use.

Reply:
We already do that ina  number of places, like when swapping shapes mode to refresh highlights only

Reply:
No. We cannot do this. In general scenario, the change of scale or units impact rendering and may require reload of data. However, in this specific case, such reload are just obsolete CPU cycles that only slowdown code execution as there are other attributes that may be modified too. So we want to have only one data reload, not more.

Reply:
It's literally identical though:

```py
layer.refresh(extent=True, thumbnail=False, data_displayed=False, highlight=False)
```

Is exactly equivalent to (and ONLY to):

```py
            self._clear_extent()
            self._clear_extent_augmented()
```

and we'd be using public api and a single source for refreshing isntead of having cache invalidation scattered all over the place.

Unless you refer to `if extent: ...` as "obsolete CPU cycles", but there's no way that's even close to being relevant here O.o

Reply:
No it is not equivalent. 
Please see this line:

https://github.com/napari/napari/blob/e44bc0164de2dfcbd82a6fab48e4af5f8d5a178d/src/napari/layers/base/base.py#L1520-L1522

Reply:
We need to always invalidate extent, and sometimes refresh view of data.

Reply:
Ok we had a call, and I understand the issue now. To summarize my understanding, the issue is that setting scale, translate etc in a layer uses property setters which will fire `refresh()` (all args as `True`). This results in many recalculations of not only the extents, but also the slicing and lots of potentially expensive calculation.

Ultimately, the *right* way to solve this is to have some nice system for delaying/queuing/deduplicating events, but most importantly to remove slicing state from layers.

Since those are both big undertakings, this PR is the best for now :)